### ISSUE TITLE
[RFE] Sticky/Customizable number of rows per page

### ISSUE THREAD

**Issue by waldner** [kinvolk/headlamp#522]
## Current situation

If the number of rows shown in a page is changed from the default 15 to some higher number, the change does not persist, so if one opens another page (or returns to the same original page afterwards), again only 15 rows are shown.

## Impact

For sections with many items, navigation can become cumbersome even with the allowed maximum of 50 rows (which however, as said, does not persist).

## Ideal future situation

The number of rows per page should be customizable to any number. The chosen number should persist (globally, or at least for the same page where it was selected). In my experience, people who like to have many items per page want to have that everywhere.

**Comment by joaquimrocha** [comment]
This is a good point. We will look into this. Thanks!

**Comment by joaquimrocha** [comment]
I have opened a PR to make the rows per page persistent. The customizable case is a bit more complicated, so we'll leave it for another time.

**Comment by waldner** [comment]
If I can suggest, and if it's not too much hassle, I'd at least add another entry to the drop-down menu where one can select the number of rows for page: "all", so you'd have eg 15, 25, 50, all.

**Comment by illume** [comment]
The PR to make them persistent was merged in #525

@waldner I like the idea of allowing a custom amount. I wonder if there is an example UI you've seen before that does something like this?

We now have a 'all' option on the notification UI pagination. Probably that would be nice here too. One problem with 'all' and persistent we have to be careful of is persisting all when there are many-many rows (enough to freeze the browser).

ps. I noticed a bug in the pagination. When changing from 15/page it showed me 1 of 39, then I switched to 50/page and the 1-39 pages numbers stayed the same.
